### PR TITLE
Remove exit modes, make default mode `exit`

### PR DESCRIPTION
**Pull request by cjappl**
- Remove all exit modes
- Move default to be `exit`
- Remove tests associated with continue mode

This has been done to simplify the v0 approach, seeing as continue has a number of edge cases and things to consider
- Not reporting the same thing multiple times
- Need to use RADSAN_OPTIONS which entails a bunch of extra gunk that makes our branch messier for v0
- Dependency on c++ headers seems a possibly problematic under some circumstances

This functionality will be brought back in at a later date

### CODE REVIEW

File: compiler-rt/lib/radsan/radsan_context.cpp [hunk 1]
```diff
@@ -46,9 +44,7 @@ void Context::expectNotRealtime(const char *intercepted_function_name) {
   if (inRealtimeContext() && !isBypassed()) {
     bypassPush();
     printDiagnostics(intercepted_function_name);
-    if (get_error_action_() == OnErrorAction::ExitWithFailure) {
-      exit(EXIT_FAILURE);
-    }
+    exit(EXIT_FAILURE);
```
Comment (davidtrevelyan) [inline compiler-rt/lib/radsan/radsan_context.cpp:47]: We sure we don't want to have an `enum class` option, e.g. `RadsanErrorMode::Exit` vs `RadsanErrorMode::Continue`, for now, or are you keen to minimise cognitive load for v0 upstreaming?

Comment (cjappl) [reply compiler-rt/lib/radsan/radsan_context.cpp:47]: I'm open to it if you feel strongly about it. I don't think it's hard to add it back in when we add the functionality back in the future, so I'd say I personally lean towards how it is. This leaning is very slight though, I defer to you. This is the cleanest total rip-out I could do (which I think has value, as opposed to a partial)

**Review by davidtrevelyan — APPROVED** [review]
